packages/frontend/src/utils/format.ts

```diff
@@ -0,0 +1,28 @@
+interface ErrorData {
+  data?: {
+    message?: string
+  }
+}
+
+export function formatError(error: Error & ErrorData) {
+  if (typeof error === 'string') {
+    return error
+  }
+
+  const { data } = error
+
+  // TODO: handle custom error messages elsewhere (and better)
+  if (data?.message === 'not enough funds for gas') {
+    return 'Insufficient funds. Please add ETH to pay for tx gas.'
```

Comment:
The native (fee) token on Polygon is MATIC and the native token on xDai id xDai, so technically "ETH" will not always be true here.